pkg/istio-agent/sds-agent.go

```diff
@@ -207,11 +200,7 @@ func NewAgent(proxyConfig *mesh.ProxyConfig, cfg *AgentConfig, sopts *security.O
 		sa.secOpts.TLSEnabled = false
 	}
 	// If proxy is using file mounted certs, JWT token is not needed.
-	if sa.secOpts.FileMountedCerts {
-		sa.secOpts.UseLocalJWT = false
-	} else {
-		sa.secOpts.UseLocalJWT = sa.CertsPath == "" // true if we don't have a key.pem
-	}
+	sa.secOpts.UseLocalJWT = !sa.FileMountedCerts
```

Comment:
That's also not good - you can't use the local JWT if it doesn't exist ( obviously ), and there are cases besides file mounted certs where a token should be used. CentralIstiod with gateway, etc.

Reply:
They are implemented.